### DISCUSSION TITLE
poll

### DISCUSSION THREAD

**Discussion by P3enguin** [zaayddev/ft_irc#5] [Q&A] [answered]
Why did you use poll  ?

**Comment by zaayddev** [comment, marked as answer]
In IRC (Internet Relay Chat) servers, polls are used to handle incoming messages from clients (i.e., users connected to the server) in an efficient and timely manner. Polling is a technique used in computer programming to check if there is any new data or events that require attention, without constantly checking or waiting for them.
In an IRC server, the poll function is used to handle the input and output events from the client connections. When the server receives a message from a client, the message is added to a queue and the poll function is called to check if there is any data waiting to be read. The poll function then waits for a specified amount of time for any input/output events to occur.
If there is data waiting to be read, the poll function reads the data and passes it to the appropriate function to process the message. This allows the server to handle multiple client connections simultaneously, without wasting resources by constantly checking for new data from each connection.
In summary, using the poll function in an IRC server helps to efficiently manage incoming messages from multiple clients, ensuring that the server can handle a large number of connections without slowing down or becoming unresponsive.
